pgmpy/prediction/NaiveAdjustmentRegressor.py

```diff
@@ -0,0 +1,307 @@
+    >>> regressor = NaiveAdjustmentRegressor(causal_graph=dag)
+    >>> regressor.fit(data[["X", "Z"]], data["Y"])
+    NaiveAdjustmentRegressor(...)
+    >>>
+    >>> # Make predictions
+    >>> predictions = regressor.predict(data[["X", "Z"]])
+    >>> print(f"Predictions shape: {predictions.shape}")
+    Predictions shape: (100,)
+
+    Using a custom estimator:
+
+    >>> from sklearn.ensemble import RandomForestRegressor
+    >>>
+    >>> # Use Random Forest as the base estimator
+    >>> rf_regressor = NaiveAdjustmentRegressor(
+    ...     causal_graph=dag,
+    ...     estimator=RandomForestRegressor(n_estimators=10, random_state=42),
+    ... )
+    >>> rf_regressor.fit(data[["X", "Z"]], data["Y"])
+    NaiveAdjustmentRegressor(...)
+
+    Example with pretreatment variables:
+
+    >>> # Create DAG with pretreatment variable P -> Y
+    >>> dag_with_pretreatment = DAG(
+    ...     ebunch=[("P", "Y"), ("Z", "X"), ("Z", "Y"), ("X", "Y")],
+    ...     roles={
+    ...         "exposure": "X",
+    ...         "outcome": "Y",
+    ...         "adjustment": ["Z"],
+    ...         "pretreatment": ["P"],
+    ...     },
+    ... )
+    >>>
+    >>> # Generate data with proper relationships using simulate
+    >>> lgbn_with_P = DAG.from_dagitty(
+    ...     "dag { P -> Y [beta=0.8] Z -> X [beta=0.5] X -> Y [beta=2.0] Z -> Y [beta=1.5] }"
+    ... )
+    >>> data_with_P = lgbn_with_P.simulate(100, seed=42)
+    >>>
+    >>> regressor_with_P = NaiveAdjustmentRegressor(causal_graph=dag_with_pretreatment)
+    >>> regressor_with_P.fit(data_with_P[["X", "Z", "P"]], data_with_P["Y"])
+    NaiveAdjustmentRegressor(...)
+    """
+
+    def __init__(
+        self,
+        causal_graph,
+        estimator: Optional[BaseEstimator] = None,
+    ):
+        self.causal_graph = causal_graph
+        self.estimator = estimator
+
+    def __sklearn_tags__(self):
+        """Tags for sklearn compatibility."""
+        tags = super().__sklearn_tags__()
+        tags.target_tags.required = True
+        tags.input_tags.allow_nan = False
+        tags.regressor_tags.poor_score = True
+        return tags
+
+    def _prepare_feature_df(
+        self, X, exposure_var=None, adjustment_vars=None, pretreatment_vars=None
+    ) -> pd.DataFrame:
+        """
+        Convert input to DataFrame and validate that column names exactly match DAG variables.
+        No column renaming/mapping - strict validation only.
+        """
+        # Step 1: Get required feature columns
+        if exposure_var is not None:
+            # Use provided parameters (during fit)
+            required_features = (
+                [exposure_var] + (adjustment_vars or []) + (pretreatment_vars or [])
+            )
+        else:
+            # Use stored instance attributes (during predict)
+            required_features = self.feature_columns_
+
+        # Step 2: Convert input to DataFrame format
+        if isinstance(X, pd.DataFrame):
+            X_df = X.copy()
```

Comment:
I don't think there is any need for this `copy`, we do not modify the dataframe any where.